### ISSUE TITLE
Filter automerging by author

### ISSUE THREAD

**Issue by Aaronontheweb** [chdsbd/kodiak#178]
Possibly related to https://github.com/chdsbd/kodiak/issues/87

I'd like to have the ability to make it more secure to operate Kodiak on OSS projects - specifically, I'd like to have the ability to filter auto-merging by _who sent the request_. 

Some use cases:

* If it's someone within my OSS organization, I'd be more lenient on automatically merging the PR than I would be if it's someone outside the organization.
* If the PR is sent by dependabot or some other trusted bot installed into our org, I'd be comfortable automatically merging those.
* In the event of a single maintainer project, the maintainer might want their own PRs merged automatically but no one else's.

**Comment by chdsbd** [comment]
Are you considering using Kodiak with `merge.require_automerge_label = false`? I think if you used automerge labels you could solve all of your use cases because only organization members can add labels and you can [configure dependabot to add labels](https://dependabot.com/docs/config-file/#default_labels) to its PRs.

If you need to use Kodiak without labels I think a few config options could be added (names TBD):

```toml
# only automerge a PR without a label if it is from a member of the GitHub organization
merge.restrict_automerge_to_org_members = true
# automerge PRs created by users in this list
merge.username_automerge_whitelist = ['dependabot']
```

I'm curious to hear your thoughts.

**Comment by chdsbd** [comment]
Based on #208 I think the simplest solution is to use GitHub branch protection to require approving reviews for a PR to merge and then setup https://github.com/hmarr/auto-approve-action to automatically approve PRs made by certain users.

**Comment by chdsbd** [comment]
This feature has been released as of #268.

An example can be found here: https://kodiakhq.com/docs/recipes#automated-dependency-updates-with-dependabot

> ## Automated dependency updates with Dependabot
> 
> While Dependabot can [automerge dependency updates](https://dependabot.com/docs/config-file/#automerged_updates), this functionality will  not work when "Required approving reviews" is configured via GitHub Branch Protection because the PR needs an approving review.
> 
> Kodiak can help us here by automatically adding an approval to all Dependabot PRs, which will allow Dependabot PRs to be automatically > merged without human intervention.
> 
> ```
> # .kodiak.toml
> version = 1
> 
> [approve]
> auto_approve_usernames = ["dependabot-preview[bot]"]
> ```

The configuration reference can be found here: https://kodiakhq.com/docs/config-reference#approveauto_approve_usernames
